docs/developer-docs/multi-chain/bitcoin/using-btc/create-transactions.mdx

```diff
@@ -0,0 +1,311 @@
+
+```
+
+<a href="https://github.com/dfinity/examples/blob/6c9e0a6b6d12978869cbf9ea2943452518909728/motoko/basic_bitcoin/src/basic_bitcoin/src/BitcoinWallet.mo#L49">
+    <div align="center">View in the full example.</div>
+</a>
+</TabItem>
+<TabItem value="rust" label="Rust">
+
+```rust
+
+let own_utxos = bitcoin_api::get_utxos(network, own_address.clone())
+    .await
+    .utxos;
+
+```
+
+<a href="https://github.com/dfinity/examples/blob/6c9e0a6b6d12978869cbf9ea2943452518909728/rust/basic_bitcoin/src/basic_bitcoin/src/bitcoin_wallet.rs#L25">
+    <div align="center">View in the full example.</div>
+</a>
+</TabItem>
+</Tabs>
+
+An UTXO has the following structure:
+
+<Tabs groupId="language">
+<TabItem value="motoko" label="Motoko" default>
+
+```motoko
+
+/// An unspent transaction output.
+public type Utxo = {
+    outpoint : OutPoint;
+    value : Satoshi;
+    height : Nat32;
+};
+
+/// A reference to a transaction output.
+public type OutPoint = {
+    txid : Blob;
+    vout : Nat32;
+};
+
+
+```
+
+</TabItem>
+<TabItem value="rust" label="Rust">
+
+```rust
+
+/// Unspent transaction output (UTXO).
+pub struct Utxo {
+    /// See [Outpoint].
+    pub outpoint: Outpoint,
+    /// Value in the units of satoshi.
+    pub value: Satoshi,
+    /// Height in the chain.
+    pub height: u32,
+}
+
+/// Identifier of [Utxo].
+pub struct Outpoint {
+    /// Transaction Identifier.
+    pub txid: Vec<u8>,
+    /// A implicit index number.
+    pub vout: u32,
+}
+
+```
+
+</TabItem>
+</Tabs>
+
+To create a transaction that sends x satoshis to a destination address, then you
+need to select a subset of the available UTXOs that cover the amount x plus the
+transaction fee.
+
+## Calculate transaction fee per byte
+
+The transaction fee of a Bitcoin transaction is calculated based on the size of
+the transaction in bytes. An appropriate fee per byte can be determined by
+looking at the fees of recent transactions that were included in the blockchain.
+The following snippet shows how to estimate the fee per byte for a transaction
+using the `bitcoin_get_current_fee_percentiles` API endpoint and choosing the
+50th percentile.
+
+<Tabs groupId="language">
+<TabItem value="motoko" label="Motoko" default>
+
+```motoko
+
+// Get fee percentiles from previous transactions to estimate our own fee.
+let fee_percentiles = await BitcoinApi.get_current_fee_percentiles(network);
+
+let fee_per_vbyte : MillisatoshiPerVByte = if(fee_percentiles.size() == 0) {
+    // There are no fee percentiles. This case can only happen on a regtest
+    // network where there are no non-coinbase transactions. In this case,
+    // we use a default of 1000 millisatoshis/vbyte (i.e. 2 satoshi/byte)
+    2000
+} else {
+    // Choose the 50th percentile for sending fees.
+    fee_percentiles[50]
+};
+
+```
+
+<a href="https://github.com/dfinity/examples/blob/6c9e0a6b6d12978869cbf9ea2943452518909728/motoko/basic_bitcoin/src/basic_bitcoin/src/BitcoinWallet.mo#L61">
```

Comment:
@jessiemongeon1 I think these links to the actual code are very useful if available as the snippets typically won't be able to run when just copying & pasting.